modules/stochastic_tools/src/samplers/ParallelSubsetSimulation.C

```diff
@@ -165,6 +168,10 @@ ParallelSubsetSimulation::sampleSetUp(const SampleMode mode)
                                _inputs_sto[j].begin() + offset + getLocalRowEnd());
     }
   }
+
+  // check if we have performed all the desired subsets
+  if (_subset == _num_subsets - 1 && sub_ind == _num_samplessub / getNumberOfRows() - 1)
```

Comment:
Not sure what the second condition is checking. Seems like just having `_subset >= _num_subsets` would suffice.

Reply:
If I use `_subset >= _num_subsets`, the termination does not happen until the next sample is taken (might be due to the order of execution). This is why I am testing if we have reached the last sample (`sub_ind == _num_samplessub / getNumberOfRows() - 1`) of the last subset (`_subset == _num_subsets - 1`).

Reply:
Oh, I understand now. That makes sense.